xir/validator.py

```diff
@@ -337,27 +340,22 @@ def _shared_definition_checks(
         declared_params: Sequence[Param],
     ) -> None:
         """Checks the names, wires, and parameters for both observable and gate definitions."""
-        if declared_wires and all(isinstance(w, str) for w in declared_wires):
-            if any(isinstance(w, int) for w in applied_wires):
-                msg = (
-                    f"Definition '{name}' is invalid. Only named wires can be applied when "
-                    "declaring named wires."
-                )
-                self._validation_messages.append(msg)
+        any_integer_applied_wires = any(isinstance(w, int) for w in applied_wires)
+        all_string_declared_wires = all(isinstance(w, str) for w in declared_wires)
 
-            if set(applied_wires) != set(declared_wires):
-                applied = ", ".join(map(str, applied_wires))
-                declared = ", ".join(map(str, declared_wires))
-                msg = (
-                    f"Definition '{name}' is invalid. Applied wires [{applied}] differ "
-                    f"from declared wires [{declared}]."
-                )
-                self._validation_messages.append(msg)
+        if any_integer_applied_wires and all_string_declared_wires:
+            msg = (
+                f"Definition '{name}' is invalid. Only named wires can be applied when "
+                "declaring named wires."
+            )
+            self._validation_messages.append(msg)
 
-        elif not all(isinstance(w, int) for w in applied_wires):
+        if not set(applied_wires).issubset(set(declared_wires)):
```

Comment:
Can we write this instead?
```suggestion
        if set(applied_wires) - set(declared_wires):
```

Reply:
Sure, that works too!

Reply:
Interesting note: `not set(applied_wires).issubset(set(declared_wires))` is actually very slightly faster than `set(applied_wires) - set(declared_wires)` (but only for large sets). Not enough to warrant a change though.

Reply:
Oh, really? I guess that makes sense because `issubset` computes a Boolean while `__sub__` constructs a set.